docs: add markdown page (#76019)
metabase/metabase: docs/dashboards/markdown.md

---

> title: Markdown in Metabase

Reviewer: ```suggestion
title: Format text with Markdown
```
Thinking people won't search for markdown (because they don't know what it is), but will search for how to format text.

> summary: Use Markdown to format text in Metabase dashboard cards, documents, and descriptions.

Reviewer: ```suggestion
summary: How to add headings, lists, code blocks, images, and more to dashboard cards and documents using Markdown syntax.
```
Just giving specific examples of things people will want to do for SEO.

---

> # Markdown in Metabase

Reviewer: ```suggestion
# Format text with Markdown
```

Use Markdown in Metabase to format text, add headings and lists, and insert links and images. In Metabase, you can use Markdown in:

- Dashboard header and text cards
- Documents

> - Descriptions

Reviewer: ```suggestion
- Descriptions (for entities likes questions and dashboards)
```

Curiously, looks like we lack descriptions for documents.

> This page covers the most common Markdown formatting options in Metabase. For a complete Markdown reference, see [The Markdown Guide](https://www.markdownguide.org/).

Reviewer: I'd put this at the bottom of the page.

```suggestion
```

> > Markdown headings and links are not the same as heading cards and link cards. This page covers Markdown *headings* (text you enlarge with `#`) and Markdown *links* (clickable text you add with brackets and parentheses). Dashboards also have heading cards and link cards, which are separate features.

Author: I will add a link to the new dashboard cards page here once I create that

Reviewer: This note should be split in two: headings, and links, and relocated to the relevant section. E.g., the note about headings should be at the bottom of the Headings sections.

In general we should try to mimize blockquotes. If they're rare, they're less likely to suffer from banner blindness.

## Headings

To create headings, use number signs (`#`). The more number signs you use, the smaller the heading. Add a space after the number signs.

```text
# Heading 1
## Heading 2
### Heading 3
```

This renders as:

![Three headings rendered in a Metabase text card](./images/markdown-headings.png)

Metabase supports six heading levels, from `#` (largest) to `######` (smallest).

## Bold and italics

To make your text bold, wrap it in two asterisks. To make your text italic, wrap it in one asterisk. To make your text bold *and* italic, wrap it in three asterisks.

```text
**This text is bold.**
*This text is italic.*

> ***This text is bold and italic.***

Reviewer: > ***This text is bold and italic.***

This didn't work. (We should also specify which flavor of markdown we support).

Author: <img width="529" height="142" alt="Image" src="https://github.com/user-attachments/assets/5ec25e2e-94db-4e02-aa69-57ae9ee003af" />

It works for me?

I added a note that we use Github Flavored Markdown in the intro section

Reviewer: Hmm. So it works in dashboard cards, but not documents...

```

This renders as:

**This text is bold.**
*This text is italic.*
***This text is bold and italic.***

## Lists

To create a bulleted list, start each line with a dash (`-`) and a space.

```text
- First item
- Second item
- Third item
```

To create a numbered list, start each line with a number, followed by a period and a space.

```text
1. First step
2. Second step
3. Third step
```

To nest a bulleted item under another bulleted item, indent the nested item with two spaces.

```text
- Fruit
  - Apple
  - Orange
- Vegetable
```

## Links

To add a link, put the link text in square brackets, followed by the URL in parentheses:

```text

> [Metabase documentation](https://www.metabase.com/docs/)

Reviewer: This link is boring. Surprise people.

```

This renders as a clickable link: [Metabase documentation](https://www.metabase.com/docs/).

> ## Images and GIFs

Reviewer: Need a screenshot here (not of data, something funny).

> To add an image, type an exclamation point (`!`), then put the alt text in square brackets, followed by the image URL in parentheses. The alt text describes the image for users with a screen reader or those who can't load the image.

Reviewer: Need to mention both that images must be reachable by URL, and the image setting. I would explain the alt text is the text between the brackets, as people reading a page on markdown probably aren't familiar with the concept, and that you're referring to the text between brackets.

```suggestion
If your admin has [allowed images from other domains](LINK), you can...
```

```text

> ![A bar chart of monthly sales](https://example.com/sales-chart.png)

Reviewer: "Sales chart" is way too serious. Don't squander this opportunity for comedy.

```

> > You can't upload an image to Metabase. Host the image somewhere else, then link to the image URL. The URL needs to be one your Metabase can access.

Reviewer: Shouldn't be a blockquote.

```suggestion
```

Also, asking people who are reading a page about markdown to host an image is a bit "draw the rest of the owl". I'd just talk about linking to image URLs.

The same syntax works for GIFs. Just use the URL of a GIF instead of an image.

## Code

To format words as inline code, wrap them in single backticks.

```text
Filter the table by the `status` column.
```

To format a block of code, wrap it in three backticks on the lines above and below the code.

````text
```
SELECT *
FROM orders
WHERE status = 'paid'
```
````

## Blockquotes

To create a blockquote, start the line with a greater than sign (`>`) and a space.

```text

> > This dashboard tracks weekly active users.

Reviewer: Another opportunity here.

```

## Horizontal lines

To separate sections of text with a horizontal line, put three dashes (`---`) on their own line. Add a blank line before and after the dashes.

```text
Section one.

---

Section two.
```

## Markdown in documents

> In a [document](../documents/introduction.md), Metabase formats your text as you type instead of showing the raw Markdown. A few things to know:

Reviewer: The autoconversion to rendered format can be a bit finicky, might want to acknowledge that.

- Dashboard header and text cards
- Documents
- Descriptions (for entities like questions and dashboards)

Metabase uses GitHub Flavored Markdown.

## Headings

To create headings, use number signs (`#`). The more number signs you use, the smaller the heading. Add a space after the number signs.

```text
# Heading 1
## Heading 2
### Heading 3
```

This renders as:

![Three headings rendered in a Metabase text card](./images/markdown-headings.png)

Metabase supports six heading levels, from `#` (largest) to `######` (smallest).

Markdown headings aren't the same as dashboard heading cards. Headings here are text you enlarge with `#`. Heading cards are a separate dashboard feature.

## Bold and italics

To make your text bold, wrap it in two asterisks. To make your text italic, wrap it in one asterisk. To make your text bold *and* italic, wrap it in three asterisks.

```text
**This text is bold.**
*This text is italic.*
***This text is bold and italic.***
```

This renders as:

**This text is bold.**
*This text is italic.*
***This text is bold and italic.***

## Lists

> To create a bulleted list, start each line with a dash (`-`) and a space.

Reviewer: ```suggestion
To create a unordered list, start each line with a dash (`-`) and a space.
```

```text
- First item
- Second item

> - Third item

Reviewer: ```suggestion
- Franz Kafka
- Jorge Luis Borges
- Ursula K. Le Guin
```

```

To create a numbered list, start each line with a number, followed by a period and a space.

```text
1. First step
2. Second step
3. Third step
```

To nest a bulleted item under another bulleted item, indent the nested item with two spaces.

```text
- Fruit
  - Apple
  - Orange
- Vegetable
```

## Links

To add a link, put the link text in square brackets, followed by the URL in parentheses:

```text
[the internet's finest webpage](https://www.metabase.com)
```

This renders as a clickable link: [the internet's finest webpage](https://www.metabase.com).

> Markdown links aren't the same as dashboard link cards, which are a separate dashboard feature for linking to other items in your Metabase. For more information, see [Link cards](../dashboards/introduction.md#link-cards).

Reviewer: ```suggestion
Markdown links aren't the same as [dashboard link cards](../dashboards/introduction.md#link-cards), which are a separate dashboard feature for linking to other items in your Metabase.
```

## Images and GIFs

> ![Cat GIF](images/markdown-cat.gif)

Reviewer: Might want to show this in the context of a dashboard, but probably fine as just bare ridiculousness.

The same syntax works for GIFs. Just use the URL of a GIF instead of an image.

## Code

To format words as inline code, wrap them in single backticks.

```text
Filter the table by the `status` column.
```

To format a block of code, wrap it in three backticks on the lines above and below the code.

````text
```
SELECT *
FROM orders
WHERE status = 'paid'
```
````

## Blockquotes

To create a blockquote, start the line with a greater than sign (`>`) and a space.

```text
> Every number is fine until someone asks how it was calculated.
```

## Horizontal lines

To separate sections of text with a horizontal line, put three dashes (`---`) on their own line. Add a blank line before and after the dashes.

```text
Section one.

---

Section two.
```

## Markdown in documents

In a [document](../documents/introduction.md), Metabase formats your text as you type instead of showing the raw Markdown. If something doesn't format the way you expect, try retyping the last character or two to trigger the conversion.

> A few other things to know:

Reviewer: ```suggestion
```
